cloudregister/registercloudguest.py

```diff
@@ -809,7 +818,10 @@ def main(args):
         os.unlink(instance_data_filepath)
 
     if registration_returncode:
-        cleanup()
+        utils.deregister_non_free_extensions()
+        utils.deregister_from_update_infrastructure()
+        utils.deregister_from_SCC()
+        utils.clean_cache()
```

Comment:
misses `/etc/hosts` cleanup

Reply:
When I refactored this the hosts cleanup was not part of it which is the reason why it's not there. Please take a closer look to the method

`def register_base_product`

This method runs in a loop `while not base_registered:` trying several times. Each time it fails it does a full cleanup. You will see that also utils.clean_hosts_file() is part of it. At the end when all attempts failed it also does the above cleanup when it already has done the cleanup for a failed attempt. As such the hosts cleanup was called in this sequence of failed attempts. That's why I did not add it here. Please verify if this is correct. In this refactor I tried to not add code logic changes

Thanks